client/components/popover/index.jsx

```diff
@@ -71,89 +71,101 @@ var Popover = React.createClass( {
 		return null;
 	},
 
-	_showOrHideTip: function( prevProps ) {
-		if ( ! this._tip ) {
-			this._tip = new Tip( '' );
-			this._tip.classname = this.props.className;
-			this._container = this._tip.inner;
+	_initTip: function() {
+		this._tip = new Tip( '' );
+		this._tip.classname = this.props.className;
+		this._container = this._tip.inner;
+	},
+
+	_renderTip: function() {
+		let content = <Content { ...omit( this.props, 'className' ) } onClickOutside={ this._onClickOutside } onClose={ this._close } />;
+
+		// Context is lost when creating a new render hierarchy, so ensure
+		// that we preserve the context that we care about
+		if ( this.context.store ) {
```

Comment:
I haven't seen this approach used before. I assume the purpose is to transfer the `Popover` component's inherited context to the DOM node created by `component-tip` which lies at the foot of `document.body`, outside of the Popover element's lineage. 

I'd love to learn what you know about this implementation and perhaps it should be addressed, but it is probably outside of the scope of this PR which concerns only the replacement of `click-outside` with `react-click-outside`. Since `component-tip` is also being replaced, perhaps it won't even be necessary or it will be eliminated during implementation of the replacement.